src/index.js

```diff
@@ -20,15 +20,82 @@ function createElement(tagName, props, ...children) {
   return element;
 }
 
-function render() {
+function render(display, calculator) {
+  const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+  const operators = [
+    { label: '+', func: (number1) => (number2) => number1 + number2 },
+    { label: '-', func: (number1) => (number2) => number1 - number2 },
+    { label: '*', func: (number1) => (number2) => number1 * number2 },
+    { label: '/', func: (number1) => (number2) => number1 / number2 },
+    { label: '=', func: () => null },
+  ];
+
+  const handleNumberClick = (event) => {
+    const input = +event.target.innerText;
+
+    if (display === 0 && !calculator) {
+      render(input);
+      return;
+    }
+
+    if (calculator && typeof calculator === 'function') {
+      const isFirstOperation = typeof calculator(display) === 'function';
+      if (isFirstOperation) {
+        render(input, calculator(display));
+        return;
+      }
+
+      const isEqualOperation = calculator() === null;
+      if (isEqualOperation) {
+        render(input);
+        return;
+      }
+
+      render(+`${display}${input}`, calculator);
+      return;
+    }
+
+    render(+`${display}${input}`);
+  };
+
+  const handleOperatorClick = (func) => {
+    if (calculator && typeof calculator === 'function') {
+      render(calculator(display), func);
+      return;
+    }
+
+    render(display, func);
+  };
+
   const element = (
     <div>
       <p>간단 계산기</p>
+      <p>{display}</p>
+      <p>
+        {numbers.map((number) => (
+          <button
+            type="button"
+            onClick={handleNumberClick}
```

Comment:
```suggestion
            onClick={() => handleClickNumber(number)}
```

로 변경하면 event에 접근할 필요가 없겠죠?

Reply:
저는 [Arrow Function in Render](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render)를 참고했었는데요.

클릭 때마다 새로운 함수가 생성되지만
전달할 매개변수가 있다면 화살표 함수를 사용해도 되겠군요.